src/backend/core/tests/documents/test_api_document_versions.py

```diff
@@ -185,6 +185,83 @@ def test_api_document_versions_list_authenticated_related_pagination(
     assert content["versions"][0]["version_id"] == all_version_ids[2]
 
 
+@pytest.mark.parametrize("via", VIA)
+def test_api_document_versions_list_authenticated_related_pagination_parent(
+    via, mock_user_teams
+):
+    """
+    When a user gains access to a document's versions via an ancestor, the date of access
+    to the parent should be used to filter versions that were created prior to the
+    user gaining access to the document.
+    """
+    user = factories.UserFactory()
+
+    client = APIClient()
+    client.force_login(user)
+
+    grand_parent = factories.DocumentFactory()
+    parent = factories.DocumentFactory(parent=grand_parent)
+    document = factories.DocumentFactory(parent=parent)
+    for i in range(3):
+        document.content = f"before {i:d}"
+        document.save()
+
+    if via == USER:
+        models.DocumentAccess.objects.create(
+            document=grand_parent,
+            user=user,
+            role=random.choice(models.RoleChoices.choices)[0],
+        )
+    elif via == TEAM:
+        mock_user_teams.return_value = ["lasuite", "unknown"]
+        models.DocumentAccess.objects.create(
+            document=grand_parent,
+            team="lasuite",
+            role=random.choice(models.RoleChoices.choices)[0],
+        )
```

Comment:
I'm not a big fan of "logic" in tests, but while this reduces the number of lines a lot, I guess it's ok ^^

Reply:
As discussed IRL. In this case, it's not much code in the if/else and it avoids duplicating all our API tests to test for users and for teams. We could think of something more elegant but not in this PR :sweat_smile: